webapp/mixins/scrollToAnchor.js

```diff
@@ -0,0 +1,23 @@
+function scrollToAnchor(anchor, { checkAnchor, $scrollTo }) {
+  if (typeof checkAnchor !== 'function')
+    throw new Error(
+      'You must define `checkAnchor` on the component if you use scrollToAnchor mixin!',
+    )
+  if (!checkAnchor(anchor)) return
+  setTimeout(() => {
+    $scrollTo(anchor)
+  }, 250)
```

Comment:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-10-01T10:41:36Z" title="Tuesday, October 1st 2019, 12:41:36 pm +02:00">Oct 1, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,20 @@
+import { scrollTo } from 'vue-scrollto'
+
+export default {
+  watch: {
+    $route(to, from) {
+      const anchor = to && to.hash
+      if (!this.checkAnchor(anchor)) return
+      setTimeout(() => {
+        scrollTo(anchor, 1000)
```


@roschaefer IMO it's a good idea to add `vue-scrollto` as a module to 'nuxt.config.js' as described [here](https://github.com/rigor789/vue-scrollto#nuxtjs).

I'd pass default options:
```
{
  duration: 1000,
  offset: -100, // to compensate fixed navbar height
}
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-01T12:48:51Z" title="Tuesday, October 1st 2019, 2:48:51 pm +02:00">Oct 1, 2019</time>_

---


From my limited experience working with `nuxt.scrollBehavior`, what I observed was that it is not called if you are on the same page, only on page transitions, so maybe the weird jump is not related to that??
https://router.vuejs.org/guide/advanced/scroll-behavior.html#scroll-behavior